### PR TITLE
Improve test coverage for util.py

### PR DESCRIPTION
**Pull request by ryanlerch**
* Adds tests for getting RPM headers from koji
* Adds tests for the splitter() method in util.py
* Remove unused Singleton class
* Remove unused get_repo_tag() method

### CODE REVIEW

File: bodhi/tests/server/test_utils.py [hunk 1]
```diff
@@ -88,6 +88,21 @@ def test_sorted_builds(self):
...
+        assert splitlist[0] == "build-0.1"
+        assert splitlist[1] == "build-0.2"
+
+        splitcommastring = util.splitter("build-0.1, build-0.2")
+        print splitcommastring
+        assert splitcommastring[0] == "build-0.1"
+        assert splitcommastring[1] == "build-0.2"
+
+        splitspacestring = util.splitter("build-0.1 build-0.2")
+        print splitspacestring
```
Comment (bowlofeggs) [inline bodhi/tests/server/test_utils.py]: Let's drop the two print statements.

File: bodhi/tests/server/test_utils.py [hunk 2]
```diff
@@ -88,6 +88,21 @@ def test_sorted_builds(self):
         assert b1 == new, b1
         assert b2 == old, b2
 
+    def test_splitter(self):
+        splitlist = util.splitter(["build-0.1", "build-0.2"])
+        assert splitlist[0] == "build-0.1"
+        assert splitlist[1] == "build-0.2"
```
Comment (bowlofeggs) [inline bodhi/tests/server/test_utils.py]: For all three of these test blocks, you could replace the assertions with a one liner that also asserts that the length is 2:

```
self.assertEqual(splitlist, ['build-0.1', 'build-0.2'])
```

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
```deleting_unused_code++```

File: bodhi/tests/server/test_utils.py (hunk 2 above)
Comment (ryanlerch) [reply bodhi/tests/server/test_utils.py]: Done. I had to make the class in this file extend unittest.TestCase to be able to use assertEqual though.

File: bodhi/tests/server/test_utils.py (hunk 1 above)
Comment (ryanlerch) [reply bodhi/tests/server/test_utils.py]: Done! Sorry about leaving debugging stuff in there :/

File: bodhi/tests/server/test_utils.py (hunk 2 above)
Comment (bowlofeggs) [reply bodhi/tests/server/test_utils.py]: Oh right, I'd forgotten about that. Some day I hope to unify the tests a bit more so they at least all subclass ```unittest.TestCase```.

**Review by bowlofeggs — APPROVED** [review]
Awesome, thanks!

**Pull request merged**